File: tests/fixtures/edgecase_repo/c/module6.c
#include <stdio.h>
#include "module5.h"

struct Base6 {
    int name;
};

typedef struct Base6 Base6;

int handle_6(const char *payload) {
    validate_6(payload);
    printf("%s", payload);
    return 0;
}

int validate_6(const char *payload) {
    return payload != 0;
}

File: tests/fixtures/edgecase_repo/python/pkg/module6.py
"""Module 6 of the generated Python layer."""

from __future__ import annotations

from pkg.module5 import Service5


class Base6:
    """Base class for service 6."""


class Service6(Base6):
    """Service 6."""

    def handle(self, payload):
        """Handle a payload."""
        self.validate(payload)
        return Service5().handle(payload)

    def validate(self, payload):
        """Validate a payload."""
        return payload is not None


def helper_6(value):
    """Module-level helper."""
    return value
